core/controllers/topic_editor.py

```diff
@@ -183,8 +184,18 @@ def get(self, topic_id):
             raise self.PageNotFoundException(
                 Exception('The topic with the given id doesn\'t exist.'))
 
+        topic_dict = topic.to_dict()
+        topic_dict['uncategorized_skill_descriptions'] = (
+            skill_services.get_skill_descriptions_by_ids(
+                topic_dict['uncategorized_skill_ids']))
+
+        for subtopic in topic_dict['subtopics']:
+            subtopic['skill_descriptions'] = (
+                skill_services.get_skill_descriptions_by_ids(
+                    subtopic['skill_ids']))
+
         self.values.update({
-            'topic': topic.to_dict()
+            'topic': topic_dict
```

Comment:
What do you think about (generally) changing the key of this to topic_dict, and similarly for other stuff? I was wondering because we have Topic domain objects in the frontend and backend, but not at the seam -- we just pass a topic_dict across. I wonder whether we should thus standardize the naming to reflect that, generally.

If you agree, it's fine not to do this in this PR, but let's file a bug.

Reply:
Done, changed it to topic_dict.